core/Command/User/MoveHome.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ *
+ * @copyright Copyright (c) 2021, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OC\Core\Command\User;
+
+use InvalidArgumentException;
+use OC\User\AccountMapper;
+use OCP\IUser;
+use OCP\IUserManager;
+use RuntimeException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MoveHome extends Command {
+
+	/**
+	 * @var IUserManager
+	 */
+	private $userManager;
+
+	public function __construct(
+		IUserManager $userManager
+	) {
+		parent::__construct();
+		$this->userManager = $userManager;
+	}
+
+	protected function configure() {
+		$this->setName('user:move-home')
+			->setDescription('Move a user\'s home folder to a new location.')
+			->addArgument('user_id', InputArgument::REQUIRED, 'Id of the user whose home folder is to be moved. ')
+			->addArgument('new_location', InputArgument::REQUIRED, 'Absolute path to the parent folder of the new location of the home folder.')
+			;
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$user = $this->getUser($input);
+		$userId = $user->getUID();
+		$oldHome = $user->getHome();
+
+		$newLocation = $this->getNewLocationForUser($input, $user);
+		$output->writeln("Move $userId from $oldHome to $newLocation");
+
+		# disable user
+		$output->writeln("Disabling user ....");
+		$user->setEnabled(false);
+
+		# copy files
+		$output->writeln("Syncing files from $oldHome/ to $newLocation ...");
+		$this->rsync($oldHome, $newLocation);
+
+		# set new home folder
+		$output->writeln("Updating user home path ....");
+		$user->setHome($newLocation);
+
+		# enable user
+		$output->writeln("Enabling user ....");
+		$user->setEnabled(true);
+
+		# notify admin about necessary LDAP change
+		$output->writeln('');
+		$output->writeln("Files for user $userId have been moved to $newLocation");
+		$output->writeln("In case you are using LDAP Home Folder please update the property in your LDAP directory for this user");
+		$output->writeln('');
+
+		$output->writeln('');
+		$output->writeln("The old home folder can now be deleted (rm -rf $oldHome)");
```

Comment:
Text proposal, because logged in admins also should not have root previledges by default:
```suggestion
		$output->writeln("The old home folder can now be deleted (sudo rm -rf $oldHome)");
```

Reply:
sudo -u www-data on ubuntu/debian
sudo -u www on suse and related
sudo -u apache on redhat (if I remember correctly)

what to do in cases where sudo is not available ....


too many options - an admin knows how to execute this command on their systems

Reply:
Misunderstanding, without `-uwww`, I did not mention this...
We have Ubuntu as default in our docs and ALWAYS use sudo in cases where higher priviledges are used.